handlers/handlers.go

```diff
@@ -176,10 +180,36 @@ func (h *Handlers) Start() {
 	h.mux.Use(middleware.Recover())
 	h.mux.Use(middleware.CORSWithConfig(corsConfig))
 	h.mux.HideBanner = true
+	h.mux.HTTPErrorHandler = h.handleError
 
 	h.mux.Logger.Fatal(h.mux.Start(":8080"))
 }
 
+func (h *Handlers) handleError(err error, c echo.Context) {
+	if err == nil {
+		return
+	}
+	var httpErr *echo.HTTPError
+	if errors.As(err, &httpErr) {
+		if strings.Contains(c.Request().Header.Get("Accept"), "application/json") {
+			c.JSON(httpErr.Code, map[string]string{"error": fmt.Sprintf("%v", httpErr.Message)})
+		} else {
+			c.String(httpErr.Code, fmt.Sprintf("%v", httpErr.Message))
+		}
+		return
+	}
+	if errors.Is(err, sql.ErrNoRows) {
```

Comment:
I don't know how effective this error check will be since there are custom error's which are also "not found"

Reply:
Also, shouldn't we be handling these cases? So I'd be a fan of removing this condition, but happy to chat

Reply:
Yeah, that makes sense, we'd probably want a `echo.HTTPError` return for 404s - would need to look over the codebase but it makes sense to remove it.

Though on that note, there's quite a few places where we do `if err != nil { return echo.NewHTTPError(http.StatusInternalServerError, err) }` - what are your thoughts on changing those to just `return err` (or really `return fmt.Errorf("something descriptive: %w", err)` now that we have the custom handler?

Reply:
> Though on that note, there's quite a few places where we do `if err != nil { return echo.NewHTTPError(http.StatusInternalServerError, err) }` - what are your thoughts on changing those to just `return err` (or really `return fmt.Errorf("something descriptive: %w", err)` now that we have the custom handler?

Agreed! (I've not done that pattern before, but it makes sense), it should definitely help us handle these pickles better

Reply:
It's one thing I really like about Echo - you handle the happy path, return the raw errors, and let the top-level error handler deal with them however is appropriate.

Reply:
True true. I'd be a fan of setting a long term goal to refactor the code to handle these SQL errors and for the timbeing don't check and continue to serve a internal server error

Reply:
Makes sense, have filed https://ystv.atlassian.net/browse/SHOW-50 to track the tech debt.